session/session.go

```diff
@@ -1926,11 +1926,19 @@ func (s *session) ExecutePreparedStmt(ctx context.Context, stmtID uint32, args [
 		return nil, err
 	}
 	s.txn.onStmtStart(preparedStmt.SQLDigest.String())
+	var is infoschema.InfoSchema
+	if preparedStmt.ForUpdateRead {
+		is = domain.GetDomain(s).InfoSchema()
+	} else if preparedStmt.InfoSchema != nil {
+		is = preparedStmt.InfoSchema.(infoschema.InfoSchema)
+	} else {
+		is = s.GetInfoSchema().(infoschema.InfoSchema)
+	}
```

Comment:
You may notice that `preparedStmtExec` does not such logic, but it does. It will eventually invoke `planner.Optimize`, which does get the latest infoschema for `preparedStmt.ForUpdateRead`.